src/ZigUP/Source/utils.c

```diff
@@ -49,7 +51,38 @@ void Measure_QuickStuff(void)
 
 void Measure_Sensor(void)
 {
+  char buffer[100];
+  uint8 i;
+  
   // make new measurement depending of autodetected sensor type
-  if (TEMP_SENSOR == 1) DHT22_Measure();
-  else if (TEMP_SENSOR == 2) ds18b20_get_temp();
+  if (TEMP_SENSOR == -1) 
+    DHT22_Measure();
+  else if (TEMP_SENSOR != 0) 
+  {
+    ds18b20_start_conversion();
+    
+    i = ds18b20_current;
+    
+    EXT_Temperatures[i] = ds18b20_get_sensor_temperature(i);
+    sprintf(buffer, "%02X%02X%02X%02X%02X%02X%02X%02X: %.2f �C",
+            ds18b20_FoundROM[i][7],ds18b20_FoundROM[i][6],ds18b20_FoundROM[i][5],ds18b20_FoundROM[i][4],
+            ds18b20_FoundROM[i][3],ds18b20_FoundROM[i][2],ds18b20_FoundROM[i][1],ds18b20_FoundROM[i][0],
+            EXT_Temperatures[i]);
+    if(i==0)
+      EXT_Temperature = EXT_Temperatures[0];
+    UART_String(buffer);
+    
+    sprintf(EXT_Temperature_string, "\x01%02X%02X%02X%02X%02X%02X%02X%02X:%.2f",
+            ds18b20_FoundROM[i][7],ds18b20_FoundROM[i][6],ds18b20_FoundROM[i][5],ds18b20_FoundROM[i][4],
+            ds18b20_FoundROM[i][3],ds18b20_FoundROM[i][2],ds18b20_FoundROM[i][1],ds18b20_FoundROM[i][0],
+            EXT_Temperatures[i]);
+    EXT_Temperature_string[0] = strlen(EXT_Temperature_string) - 1;

```

Comment:
What is the reason for this first char in this string? (And is \x01 just a placeholder? For what?)

Reply:
ZCL_DATATYPE_CHAR_STR -- String : Character String (length defined in first byte)

data structure when sending strings or arrays is: [<data lenght 1 byte>, <data>]
so you can send max 254 dytes long data to cordinator. 
There is also long version ZCL_DATATYPE_LONG_CHAR_STR where 2 first bytes define string lenght.

\x01 is just in case if data is send before temperature is set first time. But seems that is not needed.